test/unit/org/apache/cassandra/db/virtual/AbstractLoggerVirtualTableTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.virtual;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import com.datastax.driver.core.Row;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.DataRange;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public abstract class AbstractLoggerVirtualTableTest<U> extends CQLTester
+{
+    protected final String keyspace = createKeyspaceName();
+
+    protected AbstractLoggerVirtualTable<U> table;
+
+    @Test
+    public void testTruncate()
+    {
+        registerTable();
+
+        int numberOfRows = 100;
+        List<LoggingEvent> loggingEvents = getLoggingEvents(numberOfRows);
+        loggingEvents.forEach(table::add);
+
+        execute(query("truncate %s"));
+
+        assertTrue(executeNet(query("select timestamp from %s")).all().isEmpty());
```

Comment:
Can we use `assertEmpty`?

Reply:
I don't see any simple way to do that because `executeNet` returns `ResultSet` from driver but `assertEmpty` accepts `UntypedResultSet`